pkg/neorpc/filters.go

```diff
@@ -126,14 +147,76 @@ func (f *NotificationFilter) Copy() *NotificationFilter {
 		res.Name = new(string)
 		*res.Name = *f.Name
 	}
+	if len(f.Parameters) != 0 {
+		res.Parameters = slices.Clone(f.Parameters)
+	}
+	f.parametersCache = nil
```

Comment:
i tried to follow your suggestion, i believe:

> Cache should be cleaned on filter's Copy.

i treat it like a new life for the struct (otherwise i do not know why somebody needs to copy something). if you copy smth, you can try to reuse the original struct and then it will be unexpected when you have copied the struct, changed `f.Parameters`, and then called `ParametersSI` with the old return values

Reply:
> Cache should be cleaned on filter's Copy.

This sentence was about the copy itself. Cache of the copy should be cleaned. It's the way how it works in NeoGo for cachable fields of transaction/block, we need to follow.

Reply:
> you can try to reuse the original struct

The original struct should be kept unchanged (including cache). The copy may be reused by the user and modified, hence it requires clean cache.

Reply:
ok, i do not mind, if this approach has become established in neo-go. dropped cache resetting